src/UnityExtension/Assets/Editor/GitHub.Unity/UI/UserSettingsView.cs

```diff
@@ -107,43 +102,53 @@ public override void OnGUI()
             EditorGUI.EndDisabledGroup();
         }
 
+        public override void OnEnable()
+        {
+            base.OnEnable();
+            userDataHasChanged = true;
+        }
+
         private void MaybeUpdateData()
         {
-            if (Repository == null)
+            if (userDataHasChanged)
             {
-                if (!String.IsNullOrEmpty(EntryPoint.Environment.GitExecutablePath))
+                userDataHasChanged = false;
+
+                if (Repository == null)
                 {
-                    if ((cachedUser == null || String.IsNullOrEmpty(cachedUser.Name)) && GitClient != null)
-                    {
-                        GitClient.GetConfigUserAndEmail().FinallyInUI((success, ex, user) => {
-                            if (success && !String.IsNullOrEmpty(user.Name) && !String.IsNullOrEmpty(user.Email))
-                            {
-                                cachedUser = user;
-
-                                userDataHasChanged = true;
-                                Redraw();
-                            }
-                        }).Start();
-                    }
+                    UpdateUserDataFromClient();
                 }
-
-                if (userDataHasChanged)
+                else
                 {
-                    newGitName = gitName = cachedUser.Name;
-                    newGitEmail = gitEmail = cachedUser.Email;
-                    userDataHasChanged = false;
+                    newGitName = gitName = Repository.User.Name;
+                    newGitEmail = gitEmail = Repository.User.Email;
                 }
-                return;
             }
+        }
 
-            userDataHasChanged = Repository.User.Name != gitName || Repository.User.Email != gitEmail;
+        private void UpdateUserDataFromClient()
+        {
+            if (String.IsNullOrEmpty(EntryPoint.Environment.GitExecutablePath))
+            {
+                return;
+            }
 
-            if (!userDataHasChanged)
+            if (GitClient == null)
+            {
                 return;
+            }
+
+            Logger.Trace("Update user data from GitClient");
 
-            userDataHasChanged = false;
-            newGitName = gitName = Repository.User.Name;
-            newGitEmail = gitEmail = Repository.User.Email;
+            GitClient.GetConfigUserAndEmail()
+                .ThenInUI((success, user) => {
+                    if (success && !String.IsNullOrEmpty(user.Name) && !String.IsNullOrEmpty(user.Email))
```

Comment:
What happens if this is empty? Do we not want to show that?

Reply:
Logged in #429